common/defs.go

```diff
@@ -211,7 +235,7 @@ func validateRules(rules Rules) []string {
 			errMsg = append(errMsg, fmt.Sprintf("Rule #%d: Invalid protocol: %s.", ruleNo, r.Protocol))
 		}
 		if r.Protocol == "tcp" {
-			r.IsStateful = true
+			rules[i].IsStateful = true
```

Comment:
`r` is a copy of the value at `rules[i]`. To change the original, you need to access the original via the slice (`rules`) and index(`i`).